src/rosdiscover/cli.py

```diff
@@ -115,32 +116,38 @@ def _observe(args) -> SystemSummary:
     return summary
 
 
-def _periodic_observe(period: int, args: argparse.Namespace) -> SystemSummary:
+def _periodic_observe(interval: int, args: argparse.Namespace) -> SystemSummary:
     config = Config.from_yaml_string(args.config)
     obs = Observer.for_container(args.container, config)
     summary = SystemSummary({})
+    stopwatch = Stopwatch()
     iterations = 0
     go = True
+    stopwatch.start()
     while go:
         try:
-            logger.info(f"Doing observation {iterations+1}")
+            obs_start = stopwatch.duration
+            iterations += 1
+            logger.info(f"Doing observation {iterations}")
             observation = obs.observe()
             summary = SystemSummary.merge(summary, observation)
-            iterations += 1
-            if 'num_iterations' in args:
-                go = iterations < args.num_iterations
-            time.sleep(period)
+            obs_end = stopwatch.duration
+            logger.debug(f"Finished observation {iterations}; took {obs_end-obs_start} seconds.")
+            if 'duration' in args:
+                go = stopwatch.duration < args.duration
+            logger.debug('Sleeping')
+            time.sleep(interval)
+            if 'duration' in args:
+                go = stopwatch.duration < args.duration
         except KeyboardInterrupt:
             go = False
-    logger.info(f"Finished observing - {iterations+1} observations in total.")
+    stopwatch.stop()
+    logger.info(f"Finished observing - {iterations+1} observations in total after {stopwatch.duration} seconds.")
```

Comment:
It would be nice to use decimal formatting (e.g., `{stopwatch.duration:.3f}`) to display the duration

Reply:
Done.